force-app/main/default/classes/JobApplicationStatusHelper.cls

```diff
@@ -0,0 +1,103 @@
+/**
+ * @description       : This helper Class will create Tasks based on the Status of the Job Application.
+ * @author            : Raghava
+ * @group             : Cloud Smiths
+ * @last modified on  : 07-20-2024
+ * @last modified by  : Raghava
+**/
+public with sharing class JobApplicationStatusHelper {
+
+    public static List<Task> tasksToBeCreated = new List<Task>();
+
+    public static void tasksBasedOnStatus(List<Job_Application__c> newJobApplications, Map<Id, Job_Application__c> oldJobApplicationsMap){
+
+        //Looping over all the Job Application records that are passed through After Update Trigger Context
+        for(Job_Application__c jobApp : newJobApplications){
+            //Check if this is an update operation and there is a change in the Status of the Job Application
+            if(oldJobApplicationsMap != null && jobApp.Status__c != oldJobApplicationsMap.get(jobApp.Id).Status__c){
+                handleStatusChange(jobApp.Status__c, jobApp);
+            }
+            //To create tasks when the Job Application is Inserted.
+            else {
+                handleStatusChange(jobApp.Status__c, jobApp);
+            }
+    }
+
+    if(!tasksToBeCreated.isEmpty()){
+        Database.insert(tasksToBeCreated, AccessLevel.SYSTEM_MODE);
+    }
+}
+
+private static void handleStatusChange(String status, Job_Application__c jobApp){
```

Comment:
See if this method and the createTask method can be bulkified. Multiple tasks are being created on each status change. This could lead to governor limit errors.

Reply:
@triden92

I do not any other way we can bulkify this code here to avoid Governor Limits other than implementing the Batch Apex. Because in the worst case scenario, with 10000 DML Rows as Governor Limit for each transaction, this code can handle upto 2000 Job Application records without failure considering the 'Applying' status as 5 tasks are created for this status. Or else, if you feel anything else, please let me know.